### PR TITLE
Reorganize directories

### PR DESCRIPTION
**Pull request by JeodC**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description

This PR organizes the root directory structure. Relevant ddio directories were moved into a `ddio` folder at root, and relevant driver directories were moved into a `drivers` folder. CMakeLists were modified and created, and building and runtimes were tested on windows. `acmlib` folder was removed, not used, and `libacm` was moved to `third_party`.

### CODE REVIEW

File: Descent3/CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@ set(HEADERS
   BriefingParse.h
   bsp.h
   buddymenu.h
+  buddymenu.h
```
Comment (Lgt2x) [inline Descent3/CMakeLists.txt:19]: duplicate?

File: linux/CMakeLists.txt [hunk 2]
```diff
@@ -4,7 +4,11 @@ set(CPPS
   lnxcon_raw.cpp
   lnxapp.cpp
   lnxcon_null.cpp
+  lnxcontroller.cpp
   lnxdata.cpp
   registry.cpp)
-#  DynXLib.cpp
 add_library(linux STATIC ${HEADERS} ${CPPS})
+add_library(lnxcontroller STATIC ${HEADERS} ${CPPS})
```
Comment (Lgt2x) [inline linux/CMakeLists.txt:11]: now lnxcontroller.cpp is in the `linux` library, you don't need `lnxcontroller` library anymore

Comment (JeodC) [reply linux/CMakeLists.txt]: Did I modify it correctly?

Comment (Lgt2x) [reply linux/CMakeLists.txt]: looks fine

**Comment by JeodC** [comment]
That line ending thing really messed some stuff up. Whenever this gets approved I'll do it on a separate branch to avoid the messy commit history here unless this is ok.

**Pull request closed without merging**